pkg/segment/segexecution.go

```diff
@@ -459,11 +461,54 @@ func ExecuteAsyncQuery(root *structs.ASTNode, aggs *structs.QueryAggregators, qi
 	}
 
 	go func() {
-		_ = executeQueryInternal(root, aggs, qid, qc, rQuery)
+		if config.IsNewQueryPipelineEnabled() {
+			_ = executePipeRespQueryInternal(root, aggs, qid, qc)
+		} else {
+			_ = executeQueryInternal(root, aggs, qid, qc, rQuery)
+		}
 	}()
 	return rQuery.StateChan, nil
 }
 
+func executePipeRespQueryInternal(root *structs.ASTNode, aggs *structs.QueryAggregators, qid uint64, qc *structs.QueryContext) *structs.NodeResult {
```

Comment:
I think you copied a lot of this function from the non-websocket searchHander.go. Can you delete it from searchHandler.go now, or is it still different enough that we need duplicate code?

Reply:
addressed in the latest commit.